### ISSUE TITLE
Analyzer throws System.NullReferenceException for short syntax of record types

### ISSUE THREAD

**Issue by artemsnezhny** [DotNetAnalyzers/StyleCopAnalyzers#3272]
C# 9 supports short declaration form for record types, example:
`public sealed record QueueCount(int Count);`

But in this case the analyzer throw following:
```
AD0001	Analyzer 'StyleCop.Analyzers.LayoutRules.SA1502ElementMustNotBeOnASingleLine' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'
AD0001	Analyzer 'StyleCop.Analyzers.LayoutRules.SA1508ClosingBracesMustNotBePrecededByBlankLine' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.
AD0001	Analyzer 'StyleCop.Analyzers.LayoutRules.SA1500BracesForMultiLineStatementsMustNotShareLine' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'
```

In the same time for the full form this is fine
```
public sealed record WorkQueueCount
{
    public int Count { get; init; }
}
```

**Comment by bkaid** [comment]
For me this seems to be happening on 1.2.0-beta.312 but worked on 1.2.0-beta.261.

**Comment by antoinebj** [comment]
FYI I encountered this for 1500 and 1508 with both beta 312 and beta 376 on a project to which I had just added short syntax records and built with `dotnet build` command line on the CI machine

`dotnet build -c Release --no-restore $DOTNET_SOLUTION_PATH /WarnAsError`
(restore was done before)

`CSC : error AD0001: Analyzer 'StyleCop.Analyzers.LayoutRules.SA1500BracesForMultiLineStatementsMustNotShareLine' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.`
`CSC : error AD0001: Analyzer 'StyleCop.Analyzers.LayoutRules.SA1508ClosingBracesMustNotBePrecededByBlankLine' threw an exception of type 'System.NullReferenceException' with message 'Object reference not set to an instance of an object.'.`

**Comment by sharwell** [comment]
@antoinebj I was not able to reproduce the issue you described. I submitted #3421 to increase test coverage of these cases. Can you file a new issue with specific steps to reproduce?

**Comment by sharwell** [comment]
@jasondentler this issue is marked as fixed in 1.2.0-beta.321. Are you still able to reproduce it in a newer release?
